plugins/kn-migration/cmd/client.go

```diff
@@ -0,0 +1,235 @@
+import (
+	"fmt"
+	"github.com/fatih/color"
+	api_errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	api_serving "knative.dev/serving/pkg/apis/serving"
+	serving_v1_api "knative.dev/serving/pkg/apis/serving/v1"
+	serving_v1_client "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1"
+)
+
+type MigrationClient interface {
+	// Create service struct from provided options
+	ConstructService(originalservice serving_v1_api.Service) *serving_v1_api.Service
+
+	// Create revision struct from provided options
+	ConstructRevision(originalrevision serving_v1_api.Revision, config_uuid types.UID) *serving_v1_api.Revision
+
+	// Check if service exists
+	ServiceExists(name string) (bool, error)
+
+	// Get a config by service name
+	GetConfig(name string) (*serving_v1_api.Configuration, error)
+
+	// Get a service by name
+	GetService(name string) (*serving_v1_api.Service, error)
+
+	// Get service list
+	ListService() (*serving_v1_api.ServiceList, error)
+
+	// Create a service
+	CreateService(service *serving_v1_api.Service) (*serving_v1_api.Service, error)
+
+	// Delete a service by name
+	DeleteService(name string) error
+
+	// Get a revision by service name
+	GetRevision(name string) (*serving_v1_api.Revision, error)
+
+	// Create a revision
+	CreateRevision(revision *serving_v1_api.Revision, config_uuid types.UID) (*serving_v1_api.Revision, error)
+
+	// Update the given revision
+	UpdateRevision(revision *serving_v1_api.Revision) error
+
+	// Get revision list by service
+	ListRevisionByService(name string) (*serving_v1_api.RevisionList, error)
+
+	// Get service list with revisions
+	PrintServiceWithRevisions(clustername string) error
+}
+
+type migrationClient struct {
+	client    serving_v1_client.ServingV1Interface
+	namespace string
+}
+
+// Create a new client facade for the provided cl.namespace
+func NewMigrationClient(client serving_v1_client.ServingV1Interface, namespace string) MigrationClient {
+	return &migrationClient{
+		client:    client,
+		namespace: namespace,
+	}
+}
+
+// Create service struct from provided options
+func (cl *migrationClient) ConstructService(originalservice serving_v1_api.Service) *serving_v1_api.Service {
+
+	service := serving_v1_api.Service{
+		ObjectMeta: originalservice.ObjectMeta,
+	}
+
+	service.ObjectMeta.Namespace = cl.namespace
+
+	service.Spec = originalservice.Spec
+	service.Spec.Template.ObjectMeta.Name = originalservice.Status.LatestCreatedRevisionName
+	service.ObjectMeta.ResourceVersion = ""
+
+	return &service
+}
+
+// Create revision struct from provided options
+func (cl *migrationClient) ConstructRevision(originalrevision serving_v1_api.Revision, config_uuid types.UID) *serving_v1_api.Revision {
+
+	revision := serving_v1_api.Revision{
+		ObjectMeta: originalrevision.ObjectMeta,
+	}
+
+	//fmt.Println("originalrevision: ", originalrevision.ObjectMeta.Labels["serving.knative.dev/configurationGeneration"])
```

Comment:
Why do you leave commented code? Looks like debugging code

Reply:
fixed in the second commit: https://github.com/knative/client-contrib/pull/6/commits/e77b98b6cebdccc7a81fa81f9c508488611a2d3b